### PR TITLE
Wrapper user implementation

### PR DESCRIPTION
**Pull request by TheOtterlord**
Implements remaining user routes & replaces axios usage with wrapper usage.

(Also adds nodemon as a dev dependency)

### CODE REVIEW

**Review by overlisted — APPROVED** [review]
lgtm 😎

**Comment by HoloPanio** [comment]
could you replace all the resources that are provided by the wrapper in the dashboard.  Also commitlint

File: dashboard/src/context/WrapperProvider.tsx [hunk 1]
```diff
@@ -19,4 +19,4 @@ export const WrapperProvider: React.FC<WrapperProviderProps> = ({ children }) =>
   );
 };
 
-export const useAxios = () => useContext(WrapperContext);
\ No newline at end of file
+export const useWrapper = () => useContext(WrapperContext);
```
Comment (goldyydev) [inline dashboard/src/context/WrapperProvider.tsx]:
```suggestion
export const useWrapper = () => useContext(WrapperContext);

```
newline :(

**Comment by TheOtterlord** [comment]
> could you replace all the resources that are provided by the wrapper in the dashboard

Like what? I'm pretty sure I covered them all

> Also commitlint

lol, even when I use the command it breaks

**Review by goldyydev — APPROVED** [review]

File: dashboard/src/modules/auth/AuthProvider.tsx [hunk 2]
```diff
@@ -1,31 +1,16 @@
+import { User } from "@duxcore/wrapper/lib/types/user";
```
Comment (HoloPanio) [inline dashboard/src/modules/auth/AuthProvider.tsx]: Have the wrapper export the types in the index file if you could

File: dashboard/src/modules/auth/login/LoginForm.tsx [hunk 3]
```diff
@@ -1,12 +1,11 @@
-import _axios, { AxiosResponse } from "axios";
+import type { TokenPair } from "@duxcore/wrapper/lib/types/user";
```
Comment (HoloPanio) [inline dashboard/src/modules/auth/login/LoginForm.tsx]: Again move the types to the wrapper index

File: dashboard/src/modules/auth/AuthProvider.tsx [hunk 4]
```diff
@@ -1,31 +1,16 @@
...
-import React, { useEffect, useMemo, useRef, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import { Preloader } from "../../components/PreLoader";
-import { useAxios } from "../../context/AxiosProvider";
-import axiosInstance from "../../lib/axiosInstance";
-import { API_BASEURL } from "../../lib/constants";
+import { useWrapper } from "../../context/WrapperProvider";
 import { extractErrors } from "../extractErrors";
 import { useHasToken } from "./useHasToken";
 import { useTokenStore } from "./useTokenStore";
```
Comment (HoloPanio) [inline dashboard/src/modules/auth/AuthProvider.tsx]: change the useTokenStore and useHasToken to the wrapper varients throughout the entire dashboard.

File: dashboard/src/modules/auth/login/LoginPage.tsx [hunk 5]
```diff
@@ -1,8 +1,9 @@
+import type { TokenPair } from "@duxcore/wrapper/lib/types/user";
 import { Form, Formik, FormikValues } from "formik";
 import { useRouter } from "next/router";
 import React, { useEffect } from "react";
+import { useWrapper } from "../../../context/WrapperProvider";
 import { useGetIntendedPath } from "../../../hooks/useGetIntendedPath";
-import { setAxiosHeader } from "../../../lib/axiosInstance";
 import { INTENDED_PATH_KEY } from "../../../lib/constants";
 import { PageComponent } from "../../../types/PageComponent";
 import { useHasToken } from "../useHasToken";
```
Comment (HoloPanio) [inline dashboard/src/modules/auth/login/LoginPage.tsx]: port from wrapper please

File: dashboard/src/modules/auth/login/LoginPage.tsx [hunk 6]
```diff
@@ -1,8 +1,9 @@
+import type { TokenPair } from "@duxcore/wrapper/lib/types/user";
```
Comment (HoloPanio) [inline dashboard/src/modules/auth/login/LoginPage.tsx]: export types to index file

**Review by HoloPanio — CHANGES_REQUESTED** [review]
Go ahead and take a look at these comments when you get the chance

**Review by HoloPanio — APPROVED** [review]
Thanks for being so receptive with the changes that I have requested, I appreciate this very much because now we can streamline the wrapper.  I'll go ahead and push this through for you!

---

Jackson R.
Project Manager and Lead Backend Developer

**Pull request merged**